pages/index.js

```diff
@@ -1,12 +1,35 @@
 import Form from "@/components/Form/Form";
 import List from "@/components/List/List";
+import FilterButtons from "@/components/FilterButtons/FilterButtons";
+import { useState } from "react";
 
 export default function HomePage({ transactions, onAddTransaction }) {
+  const [transactionFilter, setTransactionFilter] = useState("all");
+
+  function handleSetFilter(filter) {
+    setTransactionFilter(filter);
+  }
+
+  function filteredTransactions(transactions, transactionFilter) {
+    if (transactionFilter === "all") {
+      return transactions;
+    } else {
+      return transactions.filter(
+        (transaction) => transaction.type === transactionFilter
+      );
+    }
+  }
+
```

Comment:
`transactionFilter` is the state variable, this is global (i.e. accessible within the component function) and doesn't have to be specified as a parameter.

This function "does something" that filters the transactions. For this reason, it could be advantageous to rename them.


```suggestion
  function filterTransactions(transactions) {
    if (transactionFilter === "all") {
      return transactions;
    } else {
      return transactions.filter(
        (transaction) => transaction.type === transactionFilter
      );
    }
  }

```